addons/web_editor/static/src/js/editor/image_processing.js

```diff
@@ -263,12 +263,29 @@ async function applyModifications(img, dataOptions = {}) {
  * @param {String} src URL of the image to load
  * @param {HTMLImageElement} [img] img element in which to load the image
  * @returns {Promise<HTMLImageElement>} Promise that resolves to the loaded img
+ *     or a placeholder image if the src is not found.
  */
 function loadImage(src, img = new Image()) {
+    const handleImage = (source, resolve, reject) => {
+        img.addEventListener("load", () => resolve(img), {once: true});
+        img.addEventListener("error", reject, {once: true});
+        img.src = source;
+    };
+    // The server will return a placeholder image with the following src.
+    const placeholderHref = "/web/image/__odoo__unknown__src__/";
+
     return new Promise((resolve, reject) => {
-        img.addEventListener('load', () => resolve(img), {once: true});
-        img.addEventListener('error', reject, {once: true});
-        img.src = src;
+        fetch(src)
+            .then(response => {
+                if (response.status < 200 || response.status > 299) {
```

Comment:
Actually, can you explain this line please?
Why not just 404 status? This might break working cases of redirect eg?

Reply:
@rdeodoo mhhh I didn't think about redirect... :thinking: I don't want to only check 404 because we could have 400 bad request, 401 Unauthorized, ... or 500, 503, ...

I don't know what is the best practice about http response but I suppose that if the route `/somewhere/my/image` is responding a redirection message (3XX), it doesn't give the requested image directly (otherwise, what is the diff with a 200 ?) but it respond with the new route to call to get the content we are searching for no ?

Reply:
Ok, seems to indeed be working fine, could be worth a comment tho.
`fetch()` will return `{ok: true, redirected: true, status: 200}` where I would have expected `status: 301`

Note that I was referring about this flow (which actually works fine, I just tested)
- Drag & drop Text - Image snippet
- Double click the image and replace it by a local one
- Go to the backend in debug, in "Attachments" and edit the attachment of the image you uploaded, just add a URL from an external image (which is not cors protected so `fetch()` doesnt fail), like `https://upload.wikimedia.org/wikipedia/commons/f/f9/Phoenicopterus_ruber_in_S%C3%A3o_Paulo_Zoo.jpg`
![image](https://github.com/odoo/odoo/assets/30048408/025562aa-67bb-44ae-8f99-e6a9faf8bf77)
- Reload your website page, the image will now be this one
- Open the HTML Editor and replace the `data-original-src` by the `src` of the image (same url)
- Enter edit mode and click on the image
- Add a shape on the image, it will work fine:
![image](https://github.com/odoo/odoo/assets/30048408/0ec050d6-c0a1-484d-b752-4c68f07c1e95)

I thought your commit would break this flow by using the placeholder image instead of the original src in case it was a redirect, but no.

Note that this flow might seems very weird but we save external image as attachment-redirect in Odoo 16 or 17, which might somehow lead to such an image.